robottelo/decorators.py

```diff
@@ -273,12 +273,18 @@ def bz_bug_is_open(bug_id):
     :rtype: bool
 
     """
+    upstream_mode = conf.properties.get('main.upstream', 1) == '1'
     bug = None
     try:
         bug = _get_bugzilla_bug(bug_id)
     except BugFetchError as err:
         LOGGER.warning(err.message)
-    if bug is None or bug.status not in BUGZILLA_OPEN_BUG_STATUSES:
+        return False
+    if bug is None or bug.status not in BZ_OPEN_STATUSES:
+        if not upstream_mode:
+            if (bug.whiteboard and
+                    'verified in upstream' in bug.whiteboard.lower()):
+                return True
```

Comment:
Actually - can this entire block of logic be simplified? There's nested true/false/true return statements, and it's prety hard to read through.